var/spack/repos/builtin/packages/raja/package.py

```diff
@@ -13,32 +13,42 @@ class Raja(CachedCMakePackage, CudaPackage, ROCmPackage):
-    version("0.5.1", tag="v0.5.1", submodules="True")
-    version("0.5.0", tag="v0.5.0", submodules="True")
-    version("0.4.1", tag="v0.4.1", submodules="True")
-    version("0.4.0", tag="v0.4.0", submodules="True")
+    version("2023.06.1", sha256="1a26b186624d9c8ebab728842f567a75868cd9cbf364e782fe0720e54b2d30b0")
+    version("2023.06.0", sha256="934dc1e7a3e64fe1716a730c67483f224b1ed3a7fccdae25f87bd0984a2e7f41")
+    version("2022.10.5", sha256="2d4bcb90cd8997c3a2e91dd3b453ee6bb61f4eaebd125a719c34d0235d0a44a0")
+    version("2022.10.4", sha256="359eec3955722d6853d92ede98156b01e2191bf313ef1dd47940a8812355d509")
+    version("2022.10.3", sha256="63e4fd11cd84639db052dc3410cabd720d83bde68a2fe5a4e75e6577dd830819")
+    version("2022.10.2", sha256="60d119701b44b89dfaf0d28295347f68be92803d9481e3d1cb619bf20c03c029")
+    version("2022.10.1", sha256="08c50406ccc7b2290bb8632503f4c1234179df41e6060ff07794a9ceedbc1f15")
+    version("2022.10.0", sha256="05b4c7bc258a975fc6b0e1bb440f5f9e37ddcdd53b8fc1d3ae58a29881a9539e")
+    version("2022.03.1", sha256="fbdf291ee205fe01f01bedf29d2b6334235fa49992c60658736c66dc212c706c")
+    version("2022.03.0", sha256="7d3be428947e1c191f6fab0e2f276e2fa087207bc2ebd13bd17c7c9a54efccac")
+    version("0.14.1", sha256="343c091502dafd41f94a3d868821dc9f5c24aa2224147ea0b49c331172d72b82")
+    version("0.14.0", sha256="0966472b5b7eb6582296d24d910d4288ac0834d51796455a6bd2b7da3bc024d2")
+    version("0.13.0", sha256="f34f783797b589c438436cba847d94d4c8ee1cebc309417759fae99ccd6a5152")
+    version("0.12.1", sha256="e6cf53756d2273b21cdf7fb624703bf326633fad9fee3873c58fae6cf3156176")
+    version("0.12.0", sha256="4b58a67fca9020330dda9d1d08c03a650d67391f23abe69d48a07482d83e6d8f")
+    version("0.11.0", sha256="447726dc8f4c51d5a769f080b13652064bcc9eddb10b2d6014174f796aa28f39")
+    version("0.10.1", sha256="d4e0153c6f118618731ea310351f3cce2566e9d92616afe2478c3d7c090915b1")
+    version("0.10.0", sha256="654ab55f0dad594da597499a6c109b5a1995c91e844b6162f82376f2d4f1b730")
+    version("0.9.0", sha256="b251f06676d88e2fa3ef30ccac04d55655318429a9ef1e008888b901593c5b3a")
+    version("0.8.0", sha256="c397b998bb3c847c6af03e44d85a2151b76a100bb2e883f28453f3689d90d394")
+    version("0.7.0", sha256="4127598d226936c0ea8d6e0b40da1049e65fce8efd0359bd391ec4ad7e4ecdcd")
+    version("0.6.0", sha256="d4435b5ff48443ea253d41df02b6d2988645eb79590cb3be1601664d4356e654")
+    version("0.5.3", sha256="f27d05ac77c13f4304967991539c34f946b05ee2d4628c2dcd9637c94b25e56f")
+    version("0.5.2", sha256="49d8ac6061ef3cda35c00c1d76ce5f52e4de49984863fcb28217331cf5f6a190")
+    version("0.5.1", sha256="af322bc890ed763db815d19ebe1fdb3c4c2b33855b10cdeb99b95a835f93b86e")
+    version("0.5.0", sha256="fde0a29a4f4cb35a66d9c0b34170b5bde2ceff8f8291d0c9166e9e0761012f22")
+    version("0.4.1", sha256="921d286b307215e2fa61c8424ba8039b145d3c4545bc7f12d9787e82e3bc8957")
+    version("0.4.0", sha256="ec90f91568dc9125b9150463259d069ea7ba52b38814aeb024e37708f27d2c43")
```

Comment:
I cannot confirm these version sha256 as `spack checksum` reports:

```
==> Error: Could not find any remote versions for raja
```

Reply:
Not sure why but I had to edit the `url` for each of these older versions to get `spack checksum` to find them e.g., `https://github.com/LLNL/RAJA/releases/download/v0.4.0/RAJA-0.4.0.tar.gz`

Reply:
Spack automatically substitutes the version in the `url` property.  In this case the presence/absence of the `v` in the archive file is the issue in terms of URL substitution.